ortools/linear_solver/xpress_interface.cc

```diff
@@ -1589,6 +1620,32 @@ void XpressInterface::SetLpAlgorithm(int value) {
   }
 }
 
+void XpressInterface::SetStartingLpBasis(
+    const std::vector<MPSolver::BasisStatus>& variable_statuses,
+    const std::vector<MPSolver::BasisStatus>& constraint_statuses) {
+
+  initRstat.clear();
+  initCstat.clear();
+
+  std::for_each(constraint_statuses.begin(), constraint_statuses.end(),
+                [initRstat](MPSolver::BasisStatus status) { initRstat.pushback(convertToXpressBasisStatus(status)); });
```

Comment:
The size of initRstat is known beforehand. You should resize and assign elements like this. `push_back` will re-allocate memory multiple times, which can degrade performance for large inputs.
```cpp
initRstat.resize(constraint_statuses.size());
std::for_each(...
initRstat[i] = f(i);
```
Also, make sure to capture `iniRstat` by reference, or you might change a copy of it.
```